#Issue63469 Update documentation to include performance impact warnings and workaround

shashwatpandeyvns opened the pull request:
### What does this PR do?

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/63469

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

shashwatpandeyvns committed:
#Issue63469 update Docs

rawmind commented:
@shashwatpandeyvns Could you update communication guide page as well?

shashwatpandeyvns commented:
> @shashwatpandeyvns Could you update communication guide page as well?

@rawmind I believe the update for the communication guide "https://github.com/saltstack/salt-get-started/pull/36" has been merged to the master file successfully.  All thanks to you.

shashwatpandeyvns commented:
@rawmind In this one few workflows failed and a review is required as well, is there any issue with the code?

rawmind commented:
@shashwatpandeyvns thank you for the PR. As I said I'm not a part of salt team. You should wait until someone from the team make review and approve it. Unfortunately, I can't assist you to resolve GH pipelines issues.

barbaricyawps reviewed (changes requested):
@shashwatpandeyvns , thanks for your docs PR! I love to see people who are motivated to help improve the docs.

I've made a few blocking suggestions about your rST formatting and I've also made some non-blocking suggestions about phrasing that I'm hoping you'll work to resolve.

After that, I'll need a core team member to do an additional review for technical accuracy.

Thanks again for your help!

shashwatpandeyvns committed:
#63610 Suggested changes

shashwatpandeyvns commented on doc/ref/cli/salt-key.rst:
```diff
@@ -104,10 +104,16 @@ Actions
     Reject the specified public key (use --include-all to match accepted keys
     in addition to pending keys). Globs are supported.
+    .. Warning::
+    It may cause dramatic performance salt-master degradation. 
```
@barbaricyawps Thank you Ma'am for all the suggestions and for helping me with the PR.  I am learning so many things under the guidance of you great people. I have updated the changes as you requested. Please review it and recommend any changes required.

Thanks again for your help.

barbaricyawps commented:
@shashwatpandeyvns, thanks so much for being so responsive and making those changes. I will have some time on Monday to take another look at this PR and so I'll definitely review your changes in more depth at that time. Thanks again for your help!

barbaricyawps reviewed (approved):
@shashwatpandeyvns , this looks good to me! I'll get a Core team member to do a quick technical review.

Ch3LL commented on doc/ref/cli/salt-key.rst:
```diff
@@ -104,10 +104,20 @@ Actions
     Reject the specified public key (use --include-all to match accepted keys
     in addition to pending keys). Globs are supported.
+    .. warning::
+        Using globs to reject the specified public key may cause performance issues
```
Is this caused by the `glob` or deleting too many minion's keys at once?

rawmind replied:
it's not related to using globs it happens because of AES key regeneration and then minions mass re-auth. It's enough to delete/reject just one key. See https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing

cc: @barbaricyawps

Ch3LL reviewed (commented):
just a question

barbaricyawps commented:
@shashwatpandeyvns , I'm just circling back on this PR. Did you have a chance to respond to @Ch3LL 's question about your warning note?

Ch3LL commented:
pre-commit is failing here. and it looks ot be related to this change.

Ch3LL commented:
Bump ^

twangboy commented on doc/ref/cli/salt-key.rst:
```diff
@@ -104,10 +104,20 @@ Actions
+        Using globs to reject the specified public key may cause performance issues
+        because of a Salt master degredation dramatic performance salt-master degradation. 
+        See [`performance guide`](https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing) 
+        for more information about performance impacts and workarounds.
```
This needs to be reworded:
   Using globs to reject public keys may degrade Salt master performance.  
   See the [`performance guide`](https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing)  
   for details on performance impacts and possible workarounds.

twangboy commented on doc/ref/cli/salt-key.rst:
```diff
@@ -104,10 +104,20 @@ Actions
+        Using globs to reject the specified public key may cause performance issues
+        because of a Salt master degredation dramatic performance salt-master degradation. 
+        See [`performance guide`](https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing) 
+        for more information about performance impacts and workarounds.
```
This warning needs say something more along the lines of "rejecting many keys simultaneously may cause performance issues" instead of talking about globbing as there is no globbing here.

twangboy commented on doc/ref/cli/salt-key.rst:
```diff
@@ -124,10 +134,20 @@ Actions
+        Using globs to reject the specified public key may cause performance issues
+        because of a Salt master degredation dramatic performance salt-master degradation. 
+        See [`performance guide`](https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing) 
+        for more information about performance impacts and workarounds.
```
Same wording needs to be updated here

twangboy commented on doc/ref/cli/salt-key.rst:
```diff
@@ -124,10 +134,20 @@ Actions
+        Using globs to reject the specified public key may cause performance issues
+        because of a Salt master degredation dramatic performance salt-master degradation. 
+        See [`performance guide`](https://docs.saltproject.io/en/latest/topics/tutorials/intro_scale.html#too-many-minions-re-authing) 
+        for more information about performance impacts and workarounds.
```
This warning needs say something more along the lines of "deleting many keys simultaneously may cause performance issues" instead of talking about globbing as there is no globbing here.

dwoz committed:
Fix warning wording and rST formatting in salt-key.rst

- Replace broken English and duplicated 'degradation' phrase in all 4
  warning blocks with accurate, idiomatic text
- Fix markdown link syntax to rST hyperlink syntax so Sphinx renders
  clickable links
- Add blank line before each .. warning:: directive so Sphinx parses
  the admonition correctly
- Tailor each warning to its specific operation: reject vs reject-all,
  delete vs delete-all (per twangboy review 2025-04-02)
- Wrap long lines in intro_scale.rst to fit within 80 characters

Fixes #63469

dwoz committed:
Add changelog entry for PR #63610
